plutarch-extra/Plutarch/Extra/Monad.hs

```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | TermCont-related adapters for Plutarch functions.
+module Plutarch.Extra.Monad (
+  tlet,
+  tmatch,
+) where
+
+import Plutarch.Prelude
+
+-- | Like `plet` but works in a `TermCont` monad
+tlet :: Term s a -> TermCont s (Term s a)
```

Comment:
I think we should keep using the `p` as prefix for Plutarch stuff. I tend to call termcont utils like `pletC`, `pletFieldsC`, `pmatchC` etc. But feel free to choose something else - as long as it starts with `p`.